lib/src/main/java/com/dynatrace/metric/util/Metadata.java

```diff
@@ -0,0 +1,66 @@
+package com.dynatrace.metric.util;
+
+import static com.dynatrace.metric.util.MetadataConstants.Limits.MAX_DESCRIPTION_LENGTH;
+
+class Metadata {
+
+  private Metadata() {}
+
+  static String createMetadataLine(
+      String metricKey, String description, String unit, String payloadType) {
+    String normalizedDescription = null;
+    String validUnit = null;
+
+    int builderLength = 0;
+    // if description is not empty or null, try normalize
+    if (description != null && !description.isEmpty()) {
+      normalizedDescription =
+          StringValueNormalizer.normalizeStringValue(description, MAX_DESCRIPTION_LENGTH);
+      builderLength += normalizedDescription.length();
```

Comment:
@mkleinegger found another bug here: if the dimension value is an empty quoted string, normalizeStringValue will return null and the length call will throw.

Reply:
fixed in 1da0b86099a74c1fc5b896a289553b63f47ba983